bcda/utils/common_test.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type CommonTestSuite struct {
+	suite.Suite
+}
+
+// Testing the Dedup() function that removes duplicates in string slice
+func (s *CommonTestSuite) TestDedup() {
+
+	// Sample data to be used for testing
+	var sampleSlice = []string{"one", "two", "two", "three"}
+	// Results should not have 2 "two"s
+	var result = Dedup(sampleSlice)
+
+	// Iterate through the results, and only encounter "two" once
+	// If counter is 1 after the for loopp, good... anything else, not good
+	var counter = 0
+
+	for _, v := range result {
+		if v == "two" {
+			counter++
+		}
+	}
+
+	assert.Equal(s.T(), counter, 1)
```

Comment:
Can we add a couple more checks here? Verifying that `"one"` and `"three"` exist? Just to make sure our dedupe returns all results?

Testify has `assert.Contains` that you can use here. If you use that, then you can use `assert.Len` and not have to do the loop since you know that there would be three items and all of them are accounted for/ unique.